Project/src/main/resources/templates/reviewer.html

```diff
@@ -4,25 +4,72 @@
+    <script src="/js/reviewModal.js"></script>
+    <script src="/js/upload.js"></script>
 </head>
 <body>
     <header th:insert="header.html :: header"></header>
-    <table>
+    <table class="table table-bordered" id="reviewer_table">
         <thead>
-        <th>ID</th>
         <th>Filename</th>
         <th>Status</th>
         <th>Due Date</th>
+        <th>Submission Date</th>
         </thead>
         <tbody>
         <tr th:each="article: ${articles}">
-            <td th:text="${article.id}"></td>
             <td th:text="${article.fileName}"></td>
             <td th:text="${article.state}"></td>
+            <td th:text="${article.reviewDueDate}"></td>
             <td th:text="${article.date}"></td>
-            <td><a th:href="@{/review/{id}(id=${article.id})}">Submit Review</a></td>
+            <td th:switch="${article.review}">
+                <button th:case="NULL" class="btn btn-primary" data-toggle="modal" data-target="#reviewModal" th:data-id="${article.id}">Submit Review</button>
+                <div th:case="*">
+                    <a class="btn btn-primary" th:href="@{/review/{id}(id=${article.id})}">Download Review</a>
+                    <button class="btn btn-primary" data-toggle="modal" data-target="#reviewModal" th:data-id="${article.id}">Edit Submission</button>
+                </div>
+            </td>
         </tr>
         </tbody>
     </table>
+
+    <!-- Modal -->
+    <div class="modal fade" id="reviewModal" tabindex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
+        <div class="modal-dialog modal-dialog-centered" role="document">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="exampleModalLongTitle">Submit Review</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    <form id="fileUploadForm" method="POST" enctype="multipart/form-data" th:action="@{/review}">
+                        <input type="hidden" name="articleId" id="modalArticleId" />
+                        <ul>
+                            <li>
+                                <div class="input-group">
+                                    <label class="input-group-btn">
+                                <span class="btn btn-secondary">
+                                    Browse&hellip; <input id="fileLoad" type="file" name="file" style="display: none;" multiple>
+                                </span>
+                                    </label>
+                                    <input type="text"  placeholder="Select File to Upload" class="form-control" readonly>
+                                </div>
```

Comment:
This file selector logic is wrong, check out my file-upload-thing branch. It has a better bootstrap file picker. The upload.js has the logic for that in there as well.

Reply:
This is literally your code, and I used upload.js

Reply:
I know it's my code, I already fixed it. That's why I said look in my branch. It's in master now though, just merge it in and you'll see the updated stuff.

Reply:
Gotcha